anyway/views/news_flash/api.py

```diff
@@ -153,3 +231,49 @@ def single_news_flash(news_flash_id: int):
             json.dumps(news_flash_obj.serialize(), default=str), mimetype="application/json"
         )
     return Response(status=404)
+
+
+def get_supported_resolutions() -> set:
+    return set([x.name.lower() for x in BE_CONST.SUPPORTED_RESOLUTIONS])
+
+
+def get_news_flash_by_id(id: int):
+    query = db.session.query(NewsFlash)
+    news_flash_with_id = query.filter(NewsFlash.id == id).first()
+    if news_flash_with_id is None:
+        return Response(status=404)
+    if not is_news_flash_resolution_supported(news_flash_with_id):
+        return Response("News flash location not supported", 406)
+    return Response(
+        json.dumps(news_flash_with_id.serialize(), default=str), mimetype="application/json"
+    )
+
+
+def filter_by_resolutions(query, resolutions: List[str]):
+    ands = []
+    if "suburban_road" in resolutions:
+        ands.append(
+            and_(
+                NewsFlash.resolution == BE_CONST.ResolutionCategories.SUBURBAN_ROAD.value,
+                NewsFlash.road_segment_name != None,
+            )
+        )
+    if "street" in resolutions:
+        ands.append(
+            and_(
+                NewsFlash.resolution == BE_CONST.ResolutionCategories.STREET.value,
+                NewsFlash.street1_hebrew != None,
+            )
+        )
+    if len(ands) > 1:
+        return query.filter(or_(*ands))
+    return query.filter(ands.pop())
+
+
+def normalize_query_param(key, value):
+    return value if key == "source" or key == "resolution" else value[0]
+
+
+def normalize_query(params):
+    params_non_flat = params.to_dict(flat=False)
+    return {k: normalize_query_param(k, v) for k, v in params_non_flat.items()}
```

Comment:
I find this code hard to understand

In addition, I prefer to use type hints on all new functions